src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -404,26 +406,53 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
         """
         For no-status update, we only support the following parameters:
          1. RequestPriority
-         2. Global workqueue statistics, while acquiring a workflow
+         2. SiteWhitelist
+         3. SiteBlacklist
+         4. Global workqueue statistics, while acquiring a workflow
+        As Global workqueue statistics updates are exclusive to the rest of the
+        parameters. Meaning if it is about to be updated all the rest of the
+        request_args will be ignored.
         """
-        if 'RequestPriority' in request_args:
-            # Yes, we completely ignore any other arguments posted by the user (web UI case)
-            request_args = {'RequestPriority': request_args['RequestPriority']}
-            validate_request_priority(request_args)
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
+        report = []
+        reqArgsDiff = deepcopy(request_args)
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return reduceReport(report)
+
+        if workqueue_stat_validation(reqArgsDiff):
+            report = self.reqmgr_db_service.updateRequestStats(workload.name(), reqArgsDiff)
+            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgsDiff))
+            return report
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgsDiff)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
+                workload.setPriority(reqArgsDiff['RequestPriority'])
```

Comment:
@amaltaro  let me ask something here about this line of code (not related to the current review process, though) .... 

While working on https://github.com/dmwm/WMCore/issues/12038, I tried to follow the sequence of steps taken  while updating a request priority and figure out if we could update  both the workqueue elements and the workload priority from a single place - like the `wmspec` for example (i.e. the workload itself). And then I realized   this step is somehow redundant, because workload priority update is a step already performed intrinsically during the call of Global Queue priority update. See:   https://github.com/dmwm/WMCore/blob/803b7d55cc5eb1433add61c60f310f878bc41953/src/python/WMCore/Services/WorkQueue/WorkQueue.py#L250-L256 

Is there any particular reason why should we repeat the same step twice in a sequence ....  or did we  just happened to be living with this redundant step for sometime now and simply not noticing.... Or maybe I am wrong and I cannot grasp some minor detail, which makes a difference between those two calls to update the `wmspec` at couchDB:
 * the one  coming from the workqueue service priority update and 
 * the other from the `request._handleNoStatutsUpdate` methods?

Reply:
That is a great question! At some point in the last ~17 years, it was decided that workqueue would actually have its own copy of the workload spec (as you can see it uses self.db in here as well, which points to workqueue), while the "first" workload spec is created in reqmgr2 during the workflow creation in the system (I guess in reqmgr_workload_cache db). So yes, we need to keep the current logic as those are different updates.

Reply:
Todor not like!!!
Todor think many copies of objects many sources of concurrency